### PR TITLE
feat: infer missing metadata

### LINKED ISSUE #34: feat: infer missing metadata
**Is your feature request related to a problem? Please describe.**

Metadata is not available if SRR entries or local file paths are used as sample references, or if provided sample tables are incomplete. ZARP will not be able to run without metadata.

**Describe the solution you'd like**

- Use [HTSinfer](https://github.com/zavolanlab/htsinfer) to infer missing metadata
- Specifically, implement the sample processor plugin in `zarp.plugins.sample_processors.htsinfer` by making use of the [HTSinfer workflow](https://github.com/zavolanlab/zarp/blob/2bdf65deae5d4ffacc4b1a600d7d9ed425614255/workflow/rules/htsinfer.smk) shipped with [ZARP](https://github.com/zavolanlab/zarp) at `workflows/rules/htsinfer.smk`
- Use `zarp.snakemake.run.SnakemakeExecutor` to trigger the execution of the workflow
- Triggering of the plugin should already be implemented in `zarp.zarp.process_samples()`
- Unit tests should be implemented in `tests/plugins/sample_processors/test_htsinfer.py` (needs to be created)

**Describe alternatives you've considered**

N/A

**Additional context**

You can have a look at [`zarp.plugins.sample_fetchers.sra`](https://github.com/zavolanlab/zarp-cli/blob/61f072d36740c89d1d66043c760f288bd3faba47/zarp/plugins/sample_fetchers/sra.py) for inspiration on how to implement the plugin.

Note that this functionality should consider the `--execution-mode`. If the value is `DRY_RUN`, the HTSinfer workflow should be executed in `--dry-run` mode.

### PR DESCRIPTION
**Pull request by balajtimate**
- Implement HTSinfer plugin in `zarp.plugins.sample_processors.htsinfer`
- Add unit tests for HTSinfer workflow in `tests/plugins/sample_processors/test_htsinfer.py`
- Update `zarp/config/models.py` with HTSinfer config file model
- Update `zarp/snakemake/run.py`, forcing singularity in case of HTSinfer workflow run

Fixes #34 

### Type of change

- [X] New feature (non-breaking change which adds functionality)

### Checklist

- [X] I have performed a self-review of my own code
- [X] My code follows the existing coding style, lints and generates no new
      warnings
- [X] I have added type annotations to all function/method signatures, and I
      have added type annotations for any local variables that are non-trivial,
      potentially ambiguous or might otherwise benefit from explicit typing.
- [X] I have commented my code in hard-to-understand areas
- [X] I have added [Google-style
      docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)
      to all new modules, classes,
      methods/functions or updated previously existing ones
- [X] I have added tests that prove my fix is effective or that my feature
      works
- [X] New and existing unit tests pass locally with my changes and I have not
      reduced the code coverage relative to the previous state
- [X] I have updated any sections of the app's documentation that are affected
      by the proposed changes

### CODE REVIEW

**Review by uniqueg — CHANGES_REQUESTED** [review]
Please remove `--htsinfer-config` arg

**Comment by balajtimate** [comment]
Removed `--htsinfer-config` from `args.py`, `models.py` and the config yaml file.
Also added `DRY_RUN` execution mode compatibility to the HTSinfer plugin.

File: zarp/plugins/sample_processors/htsinfer.py [hunk 1]
```diff
@@ -57,7 +57,10 @@ def process(  # pylint: disable=arguments-differ
             exec_dir=loc,
         )
         cmd = executor.compile_command(snakefile=workflow)
-        executor.run(cmd=cmd)
+        if self.config.run.execution_mode == "DRY_RUN":
+            self.records = self.records
```
Comment (uniqueg) [inline zarp/plugins/sample_processors/htsinfer.py:61]: This now needs it's own unit test

**Review by uniqueg — CHANGES_REQUESTED** [review]
Looks good, but please add a unit test for the dry run

File: zarp/plugins/sample_processors/htsinfer.py (hunk 1 above)
Comment (balajtimate) [reply zarp/plugins/sample_processors/htsinfer.py:61]: Thanks for the heads-up, I added the unit test for the dry run

**Review by uniqueg — APPROVED** [review]

**Pull request merged**